### PR TITLE
pipewire: disable webrtc-audio-processing if unavailable

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

webrtc-audio-processing only supports specific architectures, so for unsupported architectures, we have to disable it for pipewire to build.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/pipewire/default.nix [hunk 1]
```diff
@@ -56,6 +56,14 @@
 , libselinux
 }:
 
+let
+  webrtc-audio-processing' =
+    lib.findFirst (lib.meta.availableOn stdenv.hostPlatform) null [
+      webrtc-audio-processing_1
+      webrtc-audio-processing
+    ];
```
Comment (Scrumplex) [inline pkgs/development/libraries/pipewire/default.nix:64]:
```suggestion
    lib.filter (lib.meta.availableOn stdenv.hostPlatform) [
      webrtc-audio-processing_1
      webrtc-audio-processing
    ];
```

File: pkgs/development/libraries/pipewire/default.nix [hunk 2]
```diff
@@ -121,7 +129,7 @@ stdenv.mkDerivation(finalAttrs: {
     readline
     sbc
   ] ++ (if enableSystemd then [ systemd ] else [ udev ])
-  ++ (if lib.meta.availableOn stdenv.hostPlatform webrtc-audio-processing_1 then [ webrtc-audio-processing_1 ] else [ webrtc-audio-processing ])
+  ++ lib.filter (x: x != null) [ webrtc-audio-processing' ]
```
Comment (Scrumplex) [inline pkgs/development/libraries/pipewire/default.nix:132]:
```suggestion
  ++ lib.take 1 webrtc-audio-processing'
```

File: pkgs/development/libraries/pipewire/default.nix [hunk 3]
```diff
@@ -139,6 +147,7 @@ stdenv.mkDerivation(finalAttrs: {
     (lib.mesonEnable "installed_tests" true)
     (lib.mesonOption "installed_test_prefix" (placeholder "installedTests"))
     (lib.mesonOption "libjack-path" "${placeholder "jack"}/lib")
+    (lib.mesonEnable "echo-cancel-webrtc" (webrtc-audio-processing' != null))
```
Comment (Scrumplex) [inline pkgs/development/libraries/pipewire/default.nix:150]:
```suggestion
    (lib.mesonEnable "echo-cancel-webrtc" (webrtc-audio-processing' != []))
```

**Review by Scrumplex — COMMENTED** [review]
Maybe it doesn't matter to most, but I felt like this was more readable

**Comment by alyssais** [comment]
I'm not sure I understand why one would be more readable than the other? The only difference is whether you your absent value is null or an empty list, and given that we'll only ever used at most one, I don't understand what the point would be of keeping a list around?

**Review by Scrumplex — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**